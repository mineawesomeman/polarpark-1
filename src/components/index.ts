export {default as Box} from '@app/components/general/Box/Box';
export {default as Button} from '@app/components/general/Button/Button';
export {default as ContentWrapper} from '@app/components/general/ContentWrapper/ContentWrapper';
export {default as Input} from '@app/components/general/Input/Input';
export {default as Icon} from '@app/components/general/Icon/Icon';
export {default as IconButton} from '@app/components/general/IconButton/IconButton';
export {default as Link} from '@app/components/general/Link/Link';
export {default as Placeholder} from '@app/components/general/Placeholder/Placeholder';
export {default as Section} from '@app/components/general/Section/Section';
export {default as Text} from '@app/components/general/Text/Text';

export {default as ComingSoon} from '@app/components/ui/ComingSoon/ComingSoon';
export {default as Event} from '@app/components/ui/Event/Event';
export {default as Footer} from '@app/components/ui/Footer/Footer';
export {default as Header} from '@app/components/ui/Header/Header';
export {default as ScheduleItem} from '@app/components/ui/ScheduleItem/ScheduleItem';
export {default as SignOnOptions} from '@app/components/ui/SignOnOptions/SignOnOptions';
export {default as TextDivider} from '@app/components/ui/TextDivider/TextDivider';
export {default as Ticket} from '@app/components/ui/Ticket/Ticket';
export {default as TicketPreview} from '@app/components/ui/TicketPreview/TicketPreview';
