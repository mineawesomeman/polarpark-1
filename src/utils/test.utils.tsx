import {ComponentType} from 'react';
import {ThemeProvider} from '@shopify/restyle';
import {render} from '@testing-library/react-native';
import theme from '@app/config/theme';
import '@testing-library/jest-native/extend-expect';

export interface WrapperProps {
  children: JSX.Element;
}

const Wrapper = ({children}: WrapperProps): JSX.Element => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

/* Override render from @testing-library/react-native */
const customRender = (ui: JSX.Element, options: any) =>
  render(ui, {wrapper: Wrapper as ComponentType, ...options});

/* Export the entire testing library */
export * from '@testing-library/react-native';

/* Export the custom render method */
export {customRender as render};
