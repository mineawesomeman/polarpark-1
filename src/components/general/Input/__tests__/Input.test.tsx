import React from 'react';
import {fireEvent, render} from '@app/utils/test.utils';
import {Input} from '@app/components';

describe('[Input] Rendering & State Changes', () => {
  const onChangeText = jest.fn();
  const value = 'stub value';

  it('should render correctly', () => {
    const component = render(
      <Input
        iconName="user"
        onChangeText={onChangeText}
        placeholder="label"
        value={value}
      />,
      {},
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly without an icon', () => {
    const component = render(
      <Input onChangeText={onChangeText} placeholder="label" value={value} />,
      {},
    );
    expect(component).toMatchSnapshot();
  });

  //   it('should change the borderColor to #1267CD when focused', async () => {
  //     const {getByTestId, getByPlaceholderText} = render(
  //       <Input onChangeText={onChangeText} placeholder="label" value={value} />,
  //       {},
  //     );
  //     fireEvent.press(getByPlaceholderText('label'));

  //     expect(getByTestId('input')).toHaveStyle({borderColor: '#1267CD'});
  //   });
});
