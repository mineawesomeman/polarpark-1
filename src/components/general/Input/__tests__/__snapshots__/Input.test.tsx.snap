// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Input] Rendering & State Changes should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "borderColor": "#F1F1F1",
        "borderRadius": 12,
        "borderWidth": 2,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 0,
      },
    ]
  }
  testID="input"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        Array [
          Object {
            "color": "#A8A8A8",
            "flexGrow": 0,
            "height": 24,
            "justifyContent": "center",
            "marginHorizontal": 12,
            "width": 24,
          },
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    Ó§ç
  </Text>
  <TextInput
    autoCapitalize="none"
    fontFamily="NunitoSans-Regular"
    fontSize={16}
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    placeholder="label"
    placeholderTextColor="#A8A8A8"
    secureTextEntry={false}
    style={
      Array [
        Object {},
      ]
    }
    value="stub value"
  />
</View>
`;

exports[`[Input] Rendering & State Changes should render correctly without an icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "borderColor": "#F1F1F1",
        "borderRadius": 12,
        "borderWidth": 2,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
      },
    ]
  }
  testID="input"
>
  <TextInput
    autoCapitalize="none"
    fontFamily="NunitoSans-Regular"
    fontSize={16}
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    placeholder="label"
    placeholderTextColor="#A8A8A8"
    secureTextEntry={false}
    style={
      Array [
        Object {},
      ]
    }
    value="stub value"
  />
</View>
`;
